test/tools/spec-runner/index.js

```diff
@@ -187,9 +187,12 @@ function generateTopologyTests(testSuites, testContext, filter) {
         shouldRun = false;
       }
 
-      if (shouldRun && typeof filter === 'function' && !filter(spec, this.configuration)) {
-        this.currentTest.skipReason = `filtered by custom filter passed to generateTopologyTests`;
-        shouldRun = false;
+      if (shouldRun && typeof filter === 'function') {
+        const result = filter(spec, this.configuration);
```

Comment:
Minor improvement to allow for skip reasons in predicates provided to `generateTopologyTests`.